### PR TITLE
Rec 8 images network calls

### PR DESCRIPTION
**Pull request by chris-ohh**
get api calls for ingredient, equipment, and recipe images

### CODE REVIEW

File: Recipick/Model/Networking.swift [hunk 1]
```diff
@@ -41,4 +75,47 @@ final class Networking {
...
+        completionHandler(nil, RecipickError.failedToGetImage)
+        return
+      }
+      completionHandler(image, error)
+    }
+  }
+  
+  func getRecipeImage(session: Session = URLSession.shared, id: Int, size: RecipeSize, completionHandler: @escaping (UIImage?, Error?) -> Void) {
+    
+    let urlString = "https://spoonacular.com/recipeImages/\(id)-\(size.rawValue).jpg"
```
Comment (prrenfrew) [inline Recipick/Model/Networking.swift]: Enums are a nice touch and will be helpful in creating the image URLs, but the networking layer should just take in the URL, and get the image.

File: Recipick/Model/Networking.swift [hunk 2]
```diff
@@ -41,4 +75,47 @@ final class Networking {
...
+    print(urlString)
```
Comment (prrenfrew) [inline Recipick/Model/Networking.swift]: Remove debug print statements.

File: RecipickTests/MockSession.swift [hunk 3]
```diff
@@ -7,17 +7,30 @@
...
+  let image: UIImage?
 
   init(data: Data?) {
     self.data = data
+    self.image = nil
+  }
+  
+  init(image: UIImage?) {
+    self.data = nil
+    self.image = image
```
Comment (prrenfrew) [inline RecipickTests/MockSession.swift]: You could combine these two initializers into one.

`init(data: Data? = nil, image: UIImage? = nil)`

File: RecipickTests/RecipickTests.swift [hunk 4]
```diff
@@ -35,6 +37,46 @@ final class RecipickTests: XCTestCase {
...
+  
+  func testNetworkGetRecipeImageWithoutNetwork() {
+    
+    let getExpectation = expectation(description: "getExpectation")
+    let testImageData = testImage.pngData()
+    var retrievedImage: UIImage?
+    var retrievedImageData: Data?
+    var retrievedError: Error?
+    
+    Networking.shared.getRecipeImage(session: MockSession(image: self.testImage), id: 1943, size: .largest) { (image, error) in
```
Comment (prrenfrew) [inline RecipickTests/RecipickTests.swift]: Don't use self unless necessary.

File: RecipickTests/RecipickTests.swift [hunk 5]
```diff
@@ -70,4 +130,40 @@ final class RecipickTests: XCTestCase {
...
+    var retrievedImage: UIImage?
+    var retrievedError: Error?
+    
+    Networking.shared.getImage(session: MockSession(image: nil), imageType: .ingredients, imageName: "gasoline.jpg", size: .large) { (image, error) in
+      retrievedError = error
+      retrievedImage = image
+      getExpectation.fulfill()
+    }
+    
+    waitForExpectations(timeout: 1, handler: nil)
```
Comment (prrenfrew) [inline RecipickTests/RecipickTests.swift]: Timeouts should be small, like 0.1. 1 leaves room for each test to take 1 second at worst.

File: Recipick/Model/Networking.swift (hunk 1 above)
Comment (chris-ohh) [reply Recipick/Model/Networking.swift]: simplified the getimage function

File: RecipickTests/MockSession.swift (hunk 3 above)
Comment (chris-ohh) [reply RecipickTests/MockSession.swift]: fixed

File: Recipick/Model/Networking.swift (hunk 2 above)
Comment (chris-ohh) [reply Recipick/Model/Networking.swift]: fixed

File: RecipickTests/RecipickTests.swift (hunk 4 above)
Comment (chris-ohh) [reply RecipickTests/RecipickTests.swift]: fixed

File: RecipickTests/RecipickTests.swift (hunk 5 above)
Comment (chris-ohh) [reply RecipickTests/RecipickTests.swift]: fixed

**Pull request merged**